apps/main/views.py

```diff
@@ -25,3 +24,11 @@ def get_queryset(self):
 class ProductView(DetailView):
     template_name = "product.html"
     model = Product
+
+    def get_context_data(self, **kwargs):
+        context = super(ProductView, self).get_context_data()
+        context['exist_image'] = False
+        context['size_str'] = context['object'].size_format()
+        if context['object'].exists():
```

Comment:
As it is mentioned before, this usage looks like we check the existence of the object, instead of checking image availability only.